linera-views/src/dynamo_db.rs

```diff
@@ -1018,6 +1099,40 @@ impl AdminKeyValueStore for DynamoDbStore {
     }
 }
 
+impl DynamoDbStore {
+    fn inner_clone_with_root_key(
+        &self,
+        root_key: &[u8],
+    ) -> Result<DynamoDbStoreInternal, DynamoDbStoreError> {
+        #[cfg(with_metrics)]
+        {
+            self.store
+                .store
+                .store
+                .store
+                .store
+                .store
+                .store
+                .clone_with_root_key(root_key)
+        }
+        #[cfg(not(with_metrics))]
+        {
+            self.store.store.store.store.clone_with_root_key(root_key)
+        }
+    }
+
+    fn cache_size(&self) -> usize {
+        #[cfg(with_metrics)]
+        {
+            self.store.store.store.store.store.store.store.cache_size
+        }
+        #[cfg(not(with_metrics))]
+        {
+            self.store.store.store.store.cache_size
+        }
+    }
```

Comment:
Same comments as above: We could deduplicate this, and have only half as much conditional compilation.

Reply:
Ok, done.